### PR TITLE
Enable using log4j.properties file in conf directory (#768)

### PR DESCRIPTION
**Pull request by KyleFung**
Right now, Azkaban searches for a log4j.properties file baked inside of
a jar file. This change makes the Azkaban startup script search for this
file inside of the conf directory and use it if it exists. If it does
not exist, then we will revert to using the properties file found in the
jar by default.

### CODE REVIEW

**Comment by HappyRay** [comment]
Can the AZ code itself look for the log4j config in the config directory first?

**Comment by suvodeep-pyne** [comment]
do we need to do the same for exec server ?

**Comment by KyleFung** [comment]
Yes, we can. This patch updates the startup script of solo server, exec server and web server.

File: azkaban-exec-server/src/main/bash/azkaban-executor-start.sh [hunk 1]
```diff
@@ -44,6 +44,10 @@ serverpath=`pwd`
 if [ -z $AZKABAN_OPTS ]; then
   AZKABAN_OPTS="-Xmx3G"
 fi
+# Set the log4j configuration file
```
Comment (HappyRay) [inline azkaban-exec-server/src/main/bash/azkaban-executor-start.sh]: Doesn't this code assume that the conf dir will be under the binary dir? The location can be overwritten by using the -conf parameter eg. -conf temp/conf 

What do you think about adding the conf directory to the search path? Or use PropertyConfigurator see http://stackoverflow.com/questions/7390521/change-location-of-log4j-properties  What's the implication for log4j2?

**Comment by HappyRay** [comment]
Another advantage of having the code search for the conf dir first is that it will remove the duplicate logic for the three scripts.

**Comment by HappyRay** [comment]
Discussed with Kyle off line. It's tricky to add the conf directory to the search path before log4j is initialized.

So we will go with Kyle's original design.

Thanks

**Pull request merged**